core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/StackDecorator.java

```diff
@@ -128,27 +133,60 @@ public Stack decorate(@Nonnull Stack subject, @Nonnull StackRequest request, Ide
             prepareFlexSubscription(subject, request.getFlexId());
             validate(subject);
             subject = sharedServiceConfigProvider.configureStack(subject, user);
-            checkSharedServiceStackRequirements(request);
+            checkSharedServiceStackRequirements(request, user);
         }
         return subject;
     }
 
-    private void checkSharedServiceStackRequirements(StackRequest request) {
-        if (request.getClusterToAttach() != null && !isSharedServiceRequirementsMeets(request)) {
+    private void checkSharedServiceStackRequirements(StackRequest request, IdentityUser user) {
+        if (request.getClusterToAttach() != null && !isSharedServiceRequirementsMeets(request, user)) {
             throw new BadRequestException("Shared service stack should contains both Hive RDS and Ranger RDS and a properly configured LDAP also. "
                     + "One of them may be missing");
         }
     }
 
-    private boolean isSharedServiceRequirementsMeets(StackRequest request) {
-        boolean hasConfiguredLdap = request.getClusterRequest().getLdapConfig() != null;
-        boolean hasConfiguredHiveRds = request.getClusterRequest().getRdsConfigJsons().stream()
-                .anyMatch(rdsConfigRequest -> "hive".equalsIgnoreCase(rdsConfigRequest.getType()));
-        boolean hasConfiguredRangerRds = request.getClusterRequest().getRdsConfigJsons().stream()
-                .anyMatch(rdsConfigRequest -> "ranger".equalsIgnoreCase(rdsConfigRequest.getType()));
+    private boolean isSharedServiceRequirementsMeets(StackRequest request, IdentityUser user) {
+        boolean hasConfiguredLdap = hasConfiguredLdap(request);
+        boolean hasConfiguredHiveRds = hasConfiguredRdsByType(request, user, RdsType.HIVE);
+        boolean hasConfiguredRangerRds = hasConfiguredRdsByType(request, user, RdsType.RANGER);
         return hasConfiguredHiveRds && hasConfiguredRangerRds && hasConfiguredLdap;
     }
 
+    private boolean hasConfiguredRdsByType(StackRequest request, IdentityUser user, RdsType rdsType) {
+        boolean hasConfiguredRds = false;
+        if (!request.getClusterRequest().getRdsConfigJsons().isEmpty()) {
+            hasConfiguredRds = request.getClusterRequest().getRdsConfigJsons().stream()
+                    .anyMatch(rdsConfigRequest -> rdsType.name().toLowerCase().equalsIgnoreCase(rdsConfigRequest.getType()));
+        }
+        if (!hasConfiguredRds && !request.getClusterRequest().getRdsConfigNames().isEmpty()) {
+            for (String rds : request.getClusterRequest().getRdsConfigNames()) {
+                if (rdsType.name().toLowerCase().equals(rdsConfigService.getPublicRdsConfig(rds, user).getType().toLowerCase())) {
```

Comment:
equalsIgnoreCase call would be more elegant instead of calling two toLowerCase and an one equals.
same goes for line 159 and 171